### PR TITLE
(feat): Get runner and monitor image from engine spec

### PR DESCRIPTION
**Pull request by shubhamchaudhary**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

- Get runner and monitor image from engine spec

- chaosEngine yaml should be like this :
```
apiVersion: litmuschaos.io/v1alpha1
kind: ChaosEngine
metadata:
  name: engine-nginx
  namespace: ""
spec:
  appinfo:
    appns: ""
    applabel: ""
  components:
     monitor:
        image: ""
     runner:
        image: ""
  chaosServiceAccount: litmus
  monitoring: true
  experiments:
    - name: pod-delete
      spec:
        components:
~                        
```

**What this PR does / why we need it**:

**Which issue this PR fixes**: fixes #https://github.com/litmuschaos/litmus/issues/881

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 1]
```diff
@@ -539,3 +545,14 @@ func checkMonitoring(engineReconcile *reconcileEngine, reqLogger logr.Logger) (r
 	}
 	return reconcile.Result{}, nil
 }
+
+//getImage take the runner and monitor image from engine spec
+func getImage() {
+	if engine.instance.Spec.Components.Monitor.Image == "" {
+		engine.instance.Spec.Components.Monitor.Image = "litmuschaos/chaos-exporter:ci"
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: Define the image as variable!!

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 2]
```diff
@@ -539,3 +545,14 @@ func checkMonitoring(engineReconcile *reconcileEngine, reqLogger logr.Logger) (r
...
+	}
+	if engine.instance.Spec.Components.Runner.Image == "" {
+		engine.instance.Spec.Components.Runner.Image = "litmuschaos/ansible-runner:ci"
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: Define the image as variable!!

File: pkg/apis/litmuschaos/v1alpha1/zz_generated.deepcopy.go [hunk 3]
```diff
@@ -89,6 +89,7 @@ func (in *ChaosEngineList) DeepCopyObject() runtime.Object {
 func (in *ChaosEngineSpec) DeepCopyInto(out *ChaosEngineSpec) {
 	*out = *in
 	out.Appinfo = in.Appinfo
+	out.Components = in.Components
```
Comment (chandankumar4) [inline pkg/apis/litmuschaos/v1alpha1/zz_generated.deepcopy.go]: These changes are written by you or it's auto-generated?

Comment (shubhamchaudhary) [reply pkg/apis/litmuschaos/v1alpha1/zz_generated.deepcopy.go]: autogenerated

**Comment by ksatchit** [comment]
@shubhamchaudhary , thank you! While we shall go with components as an engine spec property right now with executor & monitor info, the idea will be to push these at an experiment level to enable more flexibility to chose the chaos tool/integration of choice for different experiments executed by the chaosEngine.

**Pull request merged**